esp-hal/src/i2s/parallel.rs

```diff
@@ -220,8 +221,8 @@ where
         channel.runtime_ensure_compatible(&i2s);
         let channel = channel.degrade();
 
-        PeripheralClockControl::reset(i2s.peripheral());
-        PeripheralClockControl::enable(i2s.peripheral());
+        let guard = PeripheralGuard::new(i2s.peripheral());
```

Comment:
It seems at one point the PeripheralGuard won over manually managing clocks. Would you consider updating AES/ADC, too?